tests/Driver/FlatFileDriverTest.php

```diff
@@ -92,6 +93,43 @@ public function testPopMessage()
         }
     }
 
+    public function testPopMessageWhichPushedAfterTheInitialCollect()
+    {
+        if (defined('HHVM_VERSION')) {
+            $this->markTestSkipped('Snidel is only supported on PHP.');
+        }
+
+        $this->driver->createQueue('send-newsletter');
+        $snidel = new Snidel();
```

Comment:
A test case for this behavior requires multiprocess(pop, push). [Snidel](https://github.com/ackintosh/snidel) (depends bernard!) provides the easiest way to achieve that.

Reply:
Can we use the php internal function `pcntl_fork` for this instead of snidel?

Reply:
No real big issue, just checking if we can avoid an extra dependency :smile:

Reply:
Yes. If that's what you want, I will rewrite it. 💡

Reply:
This library looks really interesting, thanks for sharing. Nevertheless, I second @acrobat 's opinion: let's keep the dependencies at the absolute minimum if possible.